biogrid/libs/biogrid-simulator/src/lib/biogrid-state/biogrid-state-class.ts

```diff
@@ -1,28 +1,77 @@
 import * as jsgraphs from 'js-graph-algorithms';
 import {
   StateGraphVertex,
-  NewStateGraphVertex,
   StateGraph,
   StateGraphEdge,
 } from '@biogrid/grid-simulator';
 
 export class BiogridState implements StateGraph {
+  // Stores the wires which connect each grid item to each other and the distance between them
   graph: jsgraphs.WeightedDiGraph;
+  // Because the graph stores numbers as vertices, graphIndexToVertex is needed to
+  // convert between a vertex in the graph and the datastructure it represents
   graphIndexToVertex: StateGraphVertex[];
 
-  constructor(numberOfVertices: number, vertices: NewStateGraphVertex[]) {
-    this.graph = new jsgraphs.WeightedDiGraph(numberOfVertices);
-    this.graphIndexToVertex = new Array<StateGraphVertex>(numberOfVertices);
+  constructor(vertices: StateGraphVertex[]) {
+    this.graph = new jsgraphs.WeightedDiGraph(vertices.length);
+    this.graphIndexToVertex = new Array<StateGraphVertex>(vertices.length);
     vertices.forEach((vertex, i) => {
-      this.graphIndexToVertex[i] = vertex.vertex;
-      this.addAllEdges(i, vertex.edges);
+      this.graphIndexToVertex[i] = vertex;
+    });
+    vertices.forEach((vertex, i) => {
+      this.addCompletlyConnectedVertex(i, vertex);
     });
   }
 
   public getGraph() {
     return this.graph;
   }
 
+  public convertStateGraphToMST() {
+    this.graph = this.getMinimumSpanningTree(this.graph);
+  }
+
+  /**
+   * Take in a vertex and connect it to every other vertex currently present in the grid
+   */
+  private addCompletelyConnectedVertex(
```

Comment:
what will this be used for? you dont need, for example, small battery A to be connected to small battery B right?

Reply:
Oh this is used in the formation of the MST, I can just move it to be a function within the convertStateGraphToMST function

Reply:
Ah I see, thanks for explaining! SGTM